### ISSUE TITLE
Search functionality could benefit from accessibility improvements

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#14083]
The current implementation of the header search bar could benefit from some accessibility improvements.

## As a screen reader user:

- The search text field doesn't have an accompanying submit button, and it may be unclear to a user that they have to press enter to submit the form
- We could usefully implement [the search role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Search_role) to help users find the search functionality as a landmark within the page
- When a user submits the search, there is no feedback to let them know the page has changed or where to find the search results. NB: this may be addressed by https://github.com/forem/forem/issues/14001

## For all users:

- There is no submit button, and form submission can only be achieved by an implicit submit (hitting enter inside the search textfield). The implicit submit is a very useful feature, but it is a [WCAG requirement to provide a submit button for every form](https://www.w3.org/WAI/WCAG21/Techniques/html/H32). This is to allow all users a predictable way to interact with the feature. This may be particularly important for users with cognitive disabilities but benefits everyone.

## Suggested changes

- Add an explicit submit/search button
- Implement the search role
- Announce the page change with an `aria-live` (likely to be covered by #14001)

**Comment by aitchiss** [comment]
Closed by #14263
